rocAL/rocAL/source/pipeline/master_graph.cpp

```diff
@@ -320,30 +310,28 @@ MasterGraph::create_tensor(const rocalTensorInfo &info, bool is_output) {
         if (output->create_from_handle(_context) != 0)
             THROW("Cannot create the tensor from handle")
         _internal_tensor_list.push_back(output);
-
-        auto *output_ext = new rocalTensor(info);
-        if (output_ext->create_from_handle(_context) != 0)
-            THROW("Cannot create the tensor from handle")
-        _output_tensor_list.push_back(output_ext);
+        _output_tensor_list.push_back(new rocalTensor(info));   // Creating a replica of the output tensor to be returned to the user
     }
 
     return output;
 }
 
 void
-MasterGraph::set_output(Image* output_image)
+MasterGraph::set_output(rocalTensor* output_tensor)
 {
-    if(output_image->is_handle_set() == false)
+    if(output_tensor->is_handle_set() == false)
     {
-        if (output_image->create_from_handle(_context) != 0)
-                THROW("Cannot create the image from handle")
-        _output_images.push_back(output_image);
+        if (output_tensor->create_from_handle(_context) != 0)
```

Comment:
tensors are created in both set_output() and create_tensor() call. We need to make sure, it is created only once for each output.

Reply:
We will check if the handle is set and then create the output tensors here to ensure we create the output only once